rmc/utils/constraint.py

```diff
@@ -295,23 +213,178 @@ def calculate_exp_per_transcript(
     group_ht = group_ht.annotate(mu_adj=group_ht.mu_agg * model[1] + model[0])
 
     logger.info(
-        "Adjusting aggregated mutation rate with coverage correction to get expected"
-        " counts..."
+        "Adjusting aggregated mutation rates with coverage correction to get expected"
+        " counts for each grouping..."
     )
     group_ht = group_ht.annotate(
         coverage_correction=get_coverage_correction_expr(
-            group_ht.exome_coverage, group_ht.coverage_model
+            group_ht.coverage, group_ht.coverage_model
         ),
     )
     group_ht = group_ht.annotate(
-        _exp=group_ht.mu_adj * group_ht.coverage_correction,
-        mu=group_ht.mu_agg * group_ht.coverage_correction,
+        expected_grouping=group_ht.mu_adj * group_ht.coverage_correction
     )
 
-    logger.info("Getting expected counts per transcript and returning...")
-    return group_ht.group_by("transcript").aggregate(
-        expected=hl.agg.sum(group_ht._exp), mu_agg=hl.agg.sum(group_ht.mu)
+    logger.info(
+        "Annotating expected counts per allele by distributing expected counts equally"
+        " among all alleles in each grouping..."
     )
+    group_ht = group_ht.annotate(
+        expected_per_variant=group_ht.expected_grouping / group_ht.n_grouping
+    )
+    context_ht = context_ht.annotate(
+        expected=group_ht[
+            context_ht.context,
+            context_ht.ref,
+            context_ht.alt,
+            context_ht.methylation_level,
+            context_ht.annotation,
+            context_ht.modifier,
+            context_ht.transcript,
+            context_ht.coverage,
+        ].expected_per_variant,
+        coverage_correction=group_ht[
+            context_ht.context,
+            context_ht.ref,
+            context_ht.alt,
+            context_ht.methylation_level,
+            context_ht.annotation,
+            context_ht.modifier,
+            context_ht.transcript,
+            context_ht.coverage,
+        ].coverage_correction,
+    )
+    # TODO: Condense the join above to use `groupings`, without re-keying `context_ht`
```

Comment:
Hmm I've tried to do `group_ht[*context_ht.groupings]` but I don't think it works?

Reply:
Which is why I left it this way

Reply:
what about `index`? something like this: https://github.com/broadinstitute/gnomad_methods/blob/main/gnomad/utils/constraint.py#L46

Reply:
Awesome yes I forgot about this!! I was trying the list comprehension method used in that script but with the brackets instead of index which didn't work, but switching it out for index works. Thank you!!